src/main/kotlin/data/weather/datasource/WeatherApiClient.kt

```diff
@@ -1,11 +1,44 @@
 package org.damascus.data.weather.datasource
 
 import org.damascus.data.weather.dto.WeatherDto
-import org.damascus.domain.model.LocationCoordinate
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.request.*
+import io.ktor.client.request.url
+import org.damascus.data.location.dataSource.LocationDataSource
+import org.damascus.data.weather.dto.LocationDto
+import org.damascus.domain.exception.LocationNotFoundException
+
+class WeatherApiClient(
+    private val client: HttpClient,
+    private val locationDataSource: LocationDataSource
+) : WeatherDataSource {
+    companion object {
+        private const val WEATHER_BASE_URL = "https://api.open-meteo.com/v1/forecast"
+    }
```

Comment:
move private things to bottom
```suggestion
    private companion object {
       const val WEATHER_BASE_URL = "https://api.open-meteo.com/v1/forecast"
    }
```

Reply:
done